Automattic/jetpack
# Add blocked requests stats to Waf_Stats class

#### Descritpion
Adds a method to the `Waf_Stats` class for calling the wpcom endpoint responsible for generating blocked requests stats.

## Proposed changes:

* Adds a `get_blocked_requests` method
* Adds a system responsible for caching blocked requests stats
* Adds a `get_blocked_requests_from_options` method for pulling blocked requests stats from the option when the cache hasn't yet expired
* Adds a `get_blocked_requests_from_api` method to call the `wpcom/v2/sites/$site/waf-stats` endpoint when the cache is non-existent or expired
* Adds `blocked_requests` data to the Protect initial state

### Other information:


## Jetpack product discussion


- 1202232924268038-as-1203282049665716

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch, once D95622-code is deployed
* Activate Protect, and proceed to the Firewall tab
* Add a Scan upgrade and visit the WAF UI
* Log `jetpackProtectInitialState` to the Dev Tools console
* Verify that the output contains the `blocked_requests` property and that value is an array of the expected stats
* Remove the upgrade and log the initial state again
* Verify that all plan-dependent stats display a default value
* Manually disable the WAF module and log the initial state again
* Verify that the stats property displays a default value
* Validate that there is no regression in app functionality introduced
